packages/suite-desktop/src/modules/power-save-blocker.ts

```diff
@@ -0,0 +1,25 @@
+import { ipcMain, powerSaveBlocker } from 'electron';
+
+import { Module } from './index';
+
+export const init: Module = () => {
+    const { logger } = global;
+
+    let powerSaveBlockerId: number | undefined;
+    return () => {
+        // TODO:
+
+        ipcMain.handle('app/power-save-blocker', shouldBlockSavePower => {
+            const isSavePowerBlocked =
+                powerSaveBlockerId && !powerSaveBlocker.isStarted(powerSaveBlockerId);
+            if (shouldBlockSavePower && !isSavePowerBlocked) {
```

Comment:
Let's create methods for each `

And then an idea is to call it from `packages/suite-desktop/src/modules/coinjoin.ts`
```typescript
      onRequest: async (method, params) => {
                    logger.debug(SERVICE_NAME, `CoinjoinClient call ${method}`);
                    if (method === 'enable') {
                        logger.debug(SERVICE_NAME, `CoinjoinClient binary enable on port ${port}`);
                        try {
                            await coinjoinMiddleware.startOnPort(port);
                        } catch (err) {
                            logger.error(SERVICE_NAME, `Start failed: ${err.message}`);
                            throw err; // pass this error to suite toast
                        }
                    }
                    if (method === 'registerAccount') {
                        // TODO: block save mode
                    }
                    if (method === 'unregisterAccount') {
                        // TODO: stop blocking save mode
                    }
                    if (method === 'disable') {
                        // TODO: also here
                        logger.debug(SERVICE_NAME, `CoinjoinClient binary stop`);
                        coinjoinMiddleware.stop();
                    }
                    // needs type casting
                    return (client[method] as any)(...params); // bind method to instance context
                },
```

Reply:
But even if we call those from coinjoin module, it would be nice to have a desktopApi event that could be useful to stop save power when Shamir's Secret Sharing recovery for example.

Reply:
Not saying it is not nice. But does device doing recovery which is in progress need data communication with host? I could imagine it does not need that

Reply:
at least, it might be useful/nice to be able to notify UI that sleep block is in place. This is something @hynek-jina might want to discuss with design department.

Reply:
Even if the device doing recovery does not need data communication with host during the process it will need once the process is finished, and that might be not successful if the host is sleeping. But I am not sure if that is really an issue.

Reply:
> at least, it might be useful/nice to be able to notify UI that sleep block is in place.

@mroz22 I think since for now power save blocking is tied to coinjoin and Suite can derive the power save state from the coinjoin session state. 🤷‍♂️ Later it can be made more autonomous.